### PR TITLE
Fixes amount of materials gotten from deconstructing tables

### LINKED ISSUE #6031: Wooden tables give less wood back then it takes to make.
It takes 3 wooden planks to make a wood table now (2 for the frame, 1 for the top), but you only get two back. 

Tested with straight wrenching, and screw drivering then wrenching.

### PR DESCRIPTION
**Pull request by Firecage**
This PR fixes the amount of materials you get from deconstructing a table. For example you would get 2 wood from deconstructing a wooden table instead of 3(2 for frame, 1 for top). Tested it and it appears to work. Please tell me if there are more problems which I overlooked.

This also makes that it is slightly easier if someone wishes to add spesshul tables which uses more than 1 materials for the top, or more or less materials for the frame and still get the correct amount from deconstructing.

Fixes #6031 

The new commit also adds the same functionality to table_frames, since deconstructing the table frames also only gave 1 of the material instead of 2 like it was supposed to. Everything still works like intended.

### CODE REVIEW

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -333,8 +335,12 @@
...
 
 	if(destroy_type == TBL_DESTROY)
-		new framestack(src.loc)
-		new buildstack(src.loc)
+		var/frameamount = framestack
+		for(var/i = 1, i <= framestackamount, i++)
+			new frameamount(get_turf(src))
+		var/buildamount = buildstack
+		for(var/i = 1, i <= buildstackamount, i++)
+			new buildamount(get_turf(src))
```
Comment (phil235) [inline code/game/objects/structures/tables_racks.dm]: why not just `new framestack(get_turf(src))` and `new buildstack(get_turf(src))` directly ?

Comment (Firecage) [inline code/game/objects/structures/tables_racks.dm]: because I am apparently an idiot.

File: code/game/objects/structures/table_frames.dm [hunk 2]
```diff
@@ -60,16 +63,12 @@
...
 /obj/structure/table_frame/wood/attackby(var/obj/item/I, mob/user)
 	if(istype(I, /obj/item/weapon/wrench))
-		user << "<span class='notice'>Now disassembling [src].</span>"
-		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
-		if(do_after(user, 30))
-			playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
-			new /obj/item/stack/sheet/mineral/wood(src.loc)
-			qdel(src)
-			return
+	 ..()
```
Comment (phil235) [inline code/game/objects/structures/table_frames.dm]: wrong indentation here?

Comment (Firecage) [inline code/game/objects/structures/table_frames.dm]: Fuck, forgot to save methinks, since it worked when I tested. HOOONK

**Comment by Firecage** [comment]
@phil235 Okay, fixed, and it still works.

Yes, wrong indentation. used a single spacing instead of a tab accidentally...and it still worked.

**Comment by phil235** [comment]
You could remove the framestack and framestackamout vars from table altogether since table_frame already has them, and use frame.framestack and frame.framestackamount in table code.

**Pull request merged**